scripts/example_script.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import girder_client
+from girder_client import HttpError
+import json
+
+
+def print_http_error(err):
+    print ('Request Error:\nstatus:%s\nresponse%s\nurl:%s'
+           '\nmethod:%s ' % (err.status, err.responseText, err.url, err.method))
+
+
+def recurse_get_resource(client, parentFolder, resourceType,
+                         parentType='folder'):
+    # now get all folders
+    folders = None
+    folderIdList = None
+    resourceList = []
+    try:
+        folders = client.listFolder(parentFolder, parentFolderType=parentType)
+    except HttpError as err:
+        print_http_error(err)
+        return []
+    folderIdList = get_field(folders, '_id')
+
+    if resourceType is 'item' and parentType is not 'collection':
+        try:
+            data = client.listItem(parentFolder)
+
+            resourceList.extend(data)
+        except HttpError as err:
+            print_http_error(err)
+            return []
+    elif resourceType is 'folder':
+        resourceList.extend(folders)
+    elif resourceType is not 'item' and resourceType is not 'folder':
+        return []
+    for folderId in folderIdList:
+        resourceList.extend(
+            recurse_get_resource(client, folderId,
+                                 resourceType))
+    return resourceList
+
+
+def print_field(data, strKey):
+    for i in range(len(data)):
+        print(data[i][strKey])
+
+
+def get_field(data, strKey):
+    return [i[strKey] for i in data]
+
+
+def make_large_image(girderClient, file_id):
+    # first check if the file is already a large image
+    item_data = None
+    try:
+        item_data = girderClient.getItem(file_id)
+    except HttpError as err:
+        print_http_error(err)
+        print('bad item id ')
+    # if the item is already a large image return
+
+    if 'largeImage' in item_data:
+        return
+    try:
+        girderClient.post('item/%s/tiles' % file_id)
+    except HttpError as err:
+        print_http_error(err)
+
+
+def isEmpty(girderClient, folder):
```

Comment:
Let's name `girderClient` as `gc`, for consistency. The scopes no longer overlap.